### DISCUSSION TITLE
`(1e4*ureg.angstrom).to_compact()` returns `10 kiloangstrom` instead of `1 micrometer`

### DISCUSSION THREAD

**Discussion by doronbehar** [hgrecco/pint#2055] [General]
As the title says, here's a MWE:

```python
(ins)>>> import pint
(ins)>>> ureg = pint.UnitRegistry()
(ins)>>> (10000*ureg.angstrom).to_compact()
<Quantity(10.0, 'kiloangstrom')>
```

Whereas:

```python
(ins)>>> (1000*ureg.nanometer).to_compact()
<Quantity(1.0, 'micrometer')>
(ins)>>> 
```

**Comment by jules-ch** [comment]
You need to specify the base or convert to base units first:

```python
>>>(10000*ureg.angstrom).to_compact("m")
<Quantity(1.0, 'micrometer')>
>>> (10000*ureg.angstrom).to_base_units().to_compact()
<Quantity(1.0, 'micrometer')>

**Reply by doronbehar** [reply]
> You need to specify the base or convert to base units first:
> 
> ```python
> >>>(10000*ureg.angstrom).to_compact("m")
> <Quantity(1.0, 'micrometer')>
> >>> (10000*ureg.angstrom).to_base_units().to_compact()
> <Quantity(1.0, 'micrometer')>
> ```

These are certainly nice workarounds, but I don't think this behavior is intuitive. Why is angstrom so special that it requires the `"m"` argument of `to_compact`? Why is angstrom treated differently then `nanometer`?
